contracts/Vault.vy

```diff
@@ -846,6 +846,7 @@ def deposit(_amount: uint256 = MAX_UINT256, recipient: address = msg.sender) ->
     @return The issued Vault shares.
     """
     assert not self.emergencyShutdown  # Deposits are locked out
+    assert not (recipient in [self, ZERO_ADDRESS])
```

Comment:
Instead, add to `erc20_safe_transfer` and `erc20_safe_transferFrom` instead

Reply:
`assert recipient not in [self, ZERO_ADDRESS]` would also work in latest Vyper

Reply:
`erc20_safe_transferFrom` is used here and will fail:
https://github.com/yearn/yearn-vaults/blob/master/contracts/Vault.vy#L1718

Reply:
@banteg https://github.com/yearn/yearn-vaults/pull/288